### PR TITLE
EVO-6707 adding message about forbidden fields

### PR DESCRIPTION
**Pull request by tobster67**
* adding message about forbidden fields

### CODE REVIEW

**Comment by lapistano** [comment]
@tobster67 guess this is still work in progress? Tests are not adapted and failing ;)

File: src/Graviton/SchemaBundle/Constraint/RecordOriginConstraint.php [hunk 1]
```diff
@@ -136,19 +138,31 @@ public function checkRecordOrigin(ConstraintEventSchema $event)
 
             // so now all unimportant fields were set to null on both - they should match if rest is untouched ;-)
             if ($userObject != $storedObject) {
+                // find out what the User tried to change
+                foreach ($userObject as $fieldName => $value) {
+                    if ($value != $storedObject->$fieldName) {
```
Comment (narcoticfresh) [inline src/Graviton/SchemaBundle/Constraint/RecordOriginConstraint.php]: @tobster67 let's say we would have this in a big service like consultation - and you're just comparing the upper most levels (as you're not recursing); you will tell the user that he tried to modify `dossierData` - which in fact the field in question is maybe in `dossierData.module.data.dude.bar.baz.myField` - how does that really help? ;-)

File: src/Graviton/SchemaBundle/Constraint/RecordOriginConstraint.php [hunk 2]
```diff
@@ -136,19 +138,31 @@ public function checkRecordOrigin(ConstraintEventSchema $event)
...
-                $this->recordOriginField
+            $error = sprintf(
+                'Prohibited modification attempt on record with %s of %s.',
+                $this->recordOriginField,
+                implode(', ', $this->recordOriginBlacklist)
             );
+            // if there are recordCoreExceptions we can be more explicit
+            if (isset($this->exceptionFieldMap[$documentClass]) && !empty($changedFields)) {
+                $error.= sprintf(
+                    ' You tried to change (%s), but You can only change by recordCoreException: (%s).',
```
Comment (narcoticfresh) [inline src/Graviton/SchemaBundle/Constraint/RecordOriginConstraint.php]: `s/You/you/`, `s/by recordCoreException://` - typos and wrong casing..

File: src/Graviton/SchemaBundle/Constraint/RecordOriginConstraint.php [hunk 3]
```diff
@@ -137,24 +142,62 @@ public function checkRecordOrigin(ConstraintEventSchema $event)
...
+            if (is_object($value) || is_array($value)) {
+                $return[$fieldName] = $this->getChangedObjectPaths($value, $compareValue, $path);
+            } else {
+                if ($value!=$compareValue) {
+                    $this->changedObjectPaths[] = implode('.', $path);
+                }
+            }
+            array_pop($path);
+        }
+        return empty($return) ? null : $return;
```
Comment (lapistano) [inline src/Graviton/SchemaBundle/Constraint/RecordOriginConstraint.php]: pls be consequent in the return values.

**Pull request merged**